tools/toolchains/__init__.py

```diff
@@ -784,9 +787,39 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    @abstractmethod
+    def parse_dependencies(self, dep_path):
+        """Parse the dependency information generated by the compiler.
+
+        Positional arguments:
+        dep_path -- the path to a file generated by a previous run of the compiler
+
+        Return value:
+        A list of all source files that the dependency file indicated were dependencies
+
+        Side effects:
+        None
+        """
+        raise NotImplemented
+
     def is_not_supported_error(self, output):
         return "#error directive: [NOT_SUPPORTED]" in output
 
+    @abstractmethod
+    def parse_output(self, output):
+        """Take in compiler output and extract sinlge line warnings and errors from it.
+
+        Positional arguments:
+        output -- a string of all the messages emitted by a run of the compiler
+
+        Return value:
+        None
+
+        Side effects:
+        call self.cc_info or self.notify with a description of the event generated by the compiler
+        """
+        raise NotImplemented
```

Comment:
> Does this make sense (it does not hurt but it does not seem correct)

Yes. Explanation below.

> You can't instantiate meta class - TypeError, same is true for the concrete classes, therefore this never can get called ? 

Well, no It can be called by subclasses of mbedToolchain. See [Here](https://docs.python.org/2.7/library/abc.html#abc.abstractmethod). The particular quote that matters is:
>The abstract methods can be called using any of the normal ‘super’ call mechanisms.

I want to use the abc meta class so that instantiating these classes without the required methods is impossible and we can test for that. I want to raise the `NotImplemented` so that users of the interface know that the superclass provides them with no help on implementing things like parse output.